styles.css

```diff
@@ -33,7 +33,19 @@ img{
 }
 
 @media(max-width:600px){
-  .main{
-    background:red;
+  header{
+    order: 1;
   }
+  main{
+    order: 2;
+  }
+  aside{
+    order: 3;
+  }
+  footer{
+    order: 4;
+  }
+}
+header h1, footer h3, main h2, aside h2 {
```

Comment:
This get's the job done. Could express it more generally in terms of the question with `header>*, footer>*, main>*, aside>*`